src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js

```diff
@@ -1,15 +1,113 @@
-const Timeouts = require('../../../platform/testing/e2e/timeouts');
-const E2eHelpers = require('../../../platform/testing/e2e/helpers');
+/* eslint-disable array-callback-return */
+const Timeouts = require('platform/testing/e2e/timeouts');
+const E2eHelpers = require('platform/testing/e2e/helpers');
 const manifest = require('../manifest.json');
+const registry = require('applications/registry.json');
 
-module.exports = E2eHelpers.createE2eTest(client => {
-  client
+const passClass = 'covid-screener-results-pass';
+const failClass = 'covid-screener-results-fail';
+
+function resetBrowser(browser) {
+  browser
     .url(`${E2eHelpers.baseUrl}/covid19screen`)
-    .waitForElementVisible('body', Timeouts.normal)
-    .axeCheck('.main');
+    .waitForElementVisible('body', Timeouts.normal);
+}
+
+function testIncompleteOnLoad(browser) {
+  const incompleteClass = 'covid-screener-results-incomplete';
+  browser.expect.element(`div.${incompleteClass}`).to.be.visible;
+}
+
+function testIsStaffButton(browser, isStaff) {
+  const isStaffButton = `button[value=${isStaff}]`;
+  const firstScrollDiv = 'div[name=multi-question-form-1-scroll-element]';
+
+  browser.waitForElementVisible(isStaffButton, Timeouts.normal);
+  browser.click(isStaffButton);
+  browser.waitForElementPresent(firstScrollDiv, Timeouts.normal);
+}
+
+/* tests end state when all answers are the same, e.g. yes or no
+    @params: 
+      isStaff: yes|no - is the user a staff member
+      value: yes|no = the answer for all subsequent questions
+*/
+function testEndStateForAnswer(browser, value, isStaff) {
+  resetBrowser(browser);
+  const visibleResultClass = value === 'no' ? passClass : failClass;
+  const notPresentResultClass = value === 'no' ? failClass : passClass;
+  const buttonValue = `button[value=${value}]`;
+
+  testIsStaffButton(browser, isStaff);
+
+  browser.elements('css selector', buttonValue, buttons => {
+    for (let i = 1; i < buttons.value.length; i++) {
+      browser.pause(1000).elementIdClick(buttons.value[i].ELEMENT);
+    }
+  });
+  browser.expect.element(`div.${visibleResultClass}`).to.be.visible;
+  browser.expect.element(`div.${notPresentResultClass}`).to.not.be.present;
+  browser.axeCheck('.main');
+}
+
+function testEndStateForMixedAnswers(browser, isStaff) {
+  resetBrowser(browser);
+
+  testIsStaffButton(browser, isStaff);
 
-  client.end();
+  browser.elements('css selector', `button[value=no]`, links => {
+    for (let i = 2; i < links.value.length; i += 2) {
+      browser.pause(1000).elementIdClick(links.value[i].ELEMENT);
```

Comment:
@bkjohnson @jhonnyoddball where do these E2E tests results get run currently, and is there any guidance about how pauses will affect build time?

Reply:
These are run as a part of our nightwatch tests which get run at the `e2e` stage in Jenkins here:

https://github.com/department-of-veterans-affairs/vets-website/blob/f29aa3a385d03479308f78c0c22c292ecae93f20/Jenkinsfile#L71-L72

I'm not aware of any specific guidance about how to use pauses.  There is some parallelization happening, so a 5 second pause in one test won't necessarily cause every other test to be slowed down by 5 seconds.

Reply:
Undocumented rule of thumb: Don't use `.pause()` unless you absolutely have to. Instead, prefer `.waitForElementVisible()` (though it looks like the elements are already visible in this case?).

With the `.pause()`, it'll not only pause a lot, slowing down every CI build / test, but will also produce flaky tests. (Often, when Jenkins gets overwhelmed, the cause of e2e test failures are arbitrary `.pause()` durations not being long enough due to the increased load on the server.)

What happens if you have no `.pause()` statements in here?